src/iguana/algorithms/clas12/RGAFiducialFilter/Algorithm.cc

```diff
@@ -0,0 +1,392 @@
+  auto& particle = GetBank(banks, b_particle, "REC::Particle");
+  auto& conf     = GetBank(banks, b_config,   "RUN::config");
+  auto* cal      = m_have_calor ? &GetBank(banks, b_calor, "REC::Calorimeter")   : nullptr;
+  auto* ft       = m_have_ft    ? &GetBank(banks, b_ft,    "REC::ForwardTagger") : nullptr;
+  auto* traj     = m_have_traj  ? &GetBank(banks, b_traj,  "REC::Traj")          : nullptr;
+
+  // prune in place
+  particle.getMutableRowList().filter([&](auto, auto row) {
+    const bool keep = Filter(static_cast<int>(row), particle, conf, cal, ft, traj);
+    return keep ? 1 : 0;
+  });
+}
+
+// core helpers
+RGAFiducialFilter::CalLayers
+RGAFiducialFilter::CollectCalHitsForTrack(const hipo::bank& cal, int pindex) {
+  CalLayers out;
+  const int n = cal.getRows();
+  for (int i=0; i<n; ++i) {
+    if (cal.getInt("pindex", i) != pindex) continue;
+    if (cal.getInt("layer",  i) != 1     ) continue; // PCal only
+    CalHit h;
+    h.sector = cal.getInt ("sector", i);
+    h.lv     = cal.getFloat("lv", i);
+    h.lw     = cal.getFloat("lw", i);
+    h.lu     = cal.getFloat("lu", i);
+    out.L1.push_back(h);
+    out.has_any = true;
+  }
+  return out;
+}
+
+bool RGAFiducialFilter::PassCalStrictness(const CalLayers& H, int strictness) {
+  if (!H.has_any) return true;
+
+  float min_lv = std::numeric_limits<float>::infinity();
+  float min_lw = std::numeric_limits<float>::infinity();
+  for (auto const& hit : H.L1) {
+    if (std::isfinite(hit.lv) && hit.lv < min_lv) min_lv = hit.lv;
+    if (std::isfinite(hit.lw) && hit.lw < min_lw) min_lw = hit.lw;
+  }
+
+  const float thr = (strictness==1 ? 9.0f : strictness==2 ? 13.5f : 18.0f);
+  return !(min_lv < thr || min_lw < thr);
+}
+
+bool RGAFiducialFilter::PassFTFiducial(int pindex, const hipo::bank* ftBank) const {
+  if (!ftBank) return true;
+
+  const auto& ft = *ftBank;
+  const int n = ft.getRows();
+  for (int i=0; i<n; ++i) {
+    if (ft.getInt("pindex", i) != pindex) continue;
+
+    const double x = ft.getFloat("x", i);
+    const double y = ft.getFloat("y", i);
+    const double r = std::hypot(x, y);
+
+    if (r < u_ft_params.rmin) return false;
+    if (r > u_ft_params.rmax) return false;
+
+    for (auto const& H : u_ft_params.holes) {
+      const double R = H[0], cx = H[1], cy = H[2];
+      const double d = std::hypot(x-cx, y-cy);
+      if (d < R) return false;
+    }
+    return true; // first associated FT hit 
+  }
+  return true; // no FT association
+}
+
+bool RGAFiducialFilter::PassCVTFiducial(int pindex, const hipo::bank* trajBank) const {
+  if (!trajBank) return true;
+
+  const auto& traj = *trajBank;
+  const int n = traj.getRows();
+
+  // One edge value per required layer for this track 
+  std::map<int, double> edge_at_layer;
+
+  double x12 = 0.0, y12 = 0.0;
+  bool saw12 = false;
+
+  for (int i = 0; i < n; ++i) {
+    if (traj.getInt("pindex", i)   != pindex) continue;
+    if (traj.getInt("detector", i) != 5     ) continue; // CVT
```

Comment:
don't hard-code detector numbers, we have [`enum DetectorType`](https://github.com/JeffersonLab/iguana/blob/7744c9d930cf2404931c94123801b86eee1e0c0a/src/iguana/algorithms/TypeDefs.h#L136-L165) to handle this; you'll probably need to `#include` the `TypeDefs.h` header file to use this.
```suggestion
    if (traj.getInt("detector", i) != DetectorType::CVT     ) continue; // CVT
```